src/MSALWrapper/TaskExecutor.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.MSALWrapper.AuthFlows
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.Logging;
+
+    /// <summary>
+    /// The task executor.
+    /// </summary>
+    internal class TaskExecutor
+    {
+        /// <summary>
+        /// The complete within.
+        /// </summary>
+        /// <typeparam name="T">The type.</typeparam>
+        /// <param name="timeout">The timeout.</param>
+        /// <param name="taskName">The task name.</param>
+        /// <param name="getTask">The get task.</param>
+        /// <param name="errorsList">The errors list.</param>
+        /// <param name="logger">The logger.</param>
+        /// <returns>The <see cref="Task"/>.</returns>
+        internal static async Task<T> CompleteWithin<T>(TimeSpan timeout, string taskName, Func<CancellationToken, Task<T>> getTask, List<Exception> errorsList = null, ILogger logger = null)

```

Comment:
Why make these optional? Anytime you are writing a new interface, it's typically better to avoid using default values, especially nulls, as this makes it far easier for null ref exceptions to creep in. (It also complicates the signature). I think it's more straight forward here to say we always need those arguments.

Reply:
This is also just my preference, but anytime I need a logger in a method, because it's such a basic part of functionality, I usually pass the ILogger as the first argument, but again this is just my preference, and is hopefully somewhat common throughout the code. (My only real reason for changing here would be for consistency).

Reply:
I agree with you here. I made the `logger` and `errorList` not nullable. I also rearranged `logger` to show first. 

I wanted to use `CompleteWithin` method for my main, first stop after CLI is invoked class to have a global timeout - AuthFlows (for reference please see this [file](https://github.com/AzureAD/microsoft-authentication-cli/pull/19/files#diff-8176dc989cffe851b038282da9231b77737a3a4d1293c2fddcedf26b66a12792) - lines 70-77

Reply:
Meaning you agree with my latest changes where `errorList` is not optional?